msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+            assertEquals("accessToken value",
+                    testCaseData.userAccountsData.get(0).getFakeAccessToken(), authenticationResult.getAccessToken());
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void performInteractiveAcquireTokenWithMockResponsesToSetupCache() {
+        TestUtils.clearCache(SHARED_PREFERENCES_NAME);
+        MockHttpClient.reset();
+        MockHttpClient.setHttpResponse(
+                MockServerResponse.getMockCloudDiscoveryResponse(),
+                HttpClient.HttpMethod.GET,
+                MockTokenCreator.CLOUD_DISCOVERY_ENDPOINT_REGEX);
+        for (UserAccountData userData: testCaseData.userAccountsData) {
+            HttpResponse mockResponseForHomeTenant = MockServerResponse.getMockTokenSuccessResponse(
+                    userData.localAccountId,
+                    userData.tenantId,
+                    userData.cloud,
+                    testCaseData.getRawClientInfo(),
+                    userData.getFakeAccessToken());
+            MockHttpClient.setHttpResponse(mockResponseForHomeTenant,
+                    HttpClient.HttpMethod.POST,
+                    userData.cloud + "/.*");
+            performInteractiveAcquireTokenCall(testCaseData.userName, userData.cloud + "/organizations");
+            if(testCaseData.homeAccountId.equals(userData.localAccountId + "." + userData.tenantId)){
+                homeAccountSignedIn = true;
+            }
+        }
+    }
+
+    @Override
+    public String[] getScopes() {
+        return USER_READ_SCOPE;
+    }
+
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().getAuthorityURL().toString();
+    }
+
+    @Override
+    public String getConfigFilePath() {
+        return TestConstants.Configurations.MULTIPLE_ACCOUNT_MODE_AAD_CONFIG_FILE_PATH;
+    }
+
+    private static class TestCaseData {
+        public final String userName;
+        public final String homeAccountId;
+        public final ArrayList<UserAccountData> userAccountsData;
+
+        public TestCaseData(
+                final String userName, String homeAccountId, ArrayList<UserAccountData> userAccountsData) {
+            this.userName = userName;
+            this.homeAccountId = homeAccountId;
+            this.userAccountsData = userAccountsData;
+        }
+        public String getRawClientInfo() {
+            final String uid = homeAccountId.split("\\.")[0];
+            final String utid = homeAccountId.split("\\.")[1];
+            final String claims = "{\"uid\":\"" + uid + "\",\"utid\":\"" + utid + "\"}";
+
+            return new String(Base64.encode(claims.getBytes(
+                    Charset.forName("UTF-8")), Base64.NO_PADDING | Base64.NO_WRAP | Base64.URL_SAFE));
+        }
+
+    }
+
+    private static class UserAccountData {
+        public final String cloud;
```

Comment:
I think you are actually passing in Authority here....may be better to rename this as authority

Reply:
done